woocommerce/woocommerce-ios
# [Blaze] Edit Ad Screen UI

Part of: #11588

## Description

Adds a new screen for editing a Blaze Ad's image, tagline and description. 

**Changes**
- Adds SwiftUI view for Blaze Edit Ad screen
- Navigates to Edit Ad screen from the campaign preview screen
- Most image picker logic is shamelessly copied from `AddProductFromImageFormImageView` and relevant files.
- We will add unit tests and other functionality in future PRs.

I'm sorry about the 500+ line changes. I added the image selection logic to make the UI flow testable.

## Testing instructions
- Log in to a store with at least one existing product and is eligible for Blaze.
- Navigate to the Products tab and select any public product.
- On the product form, select Promote with Blaze.
- Notice that the campaign creation form is presented.
- Select the "Edit Ad" button to open the new Edit Ad screen
- In the "Edit Ad" screen tap on the "Change image" button and validate that image selection works. (Test camera, device and media library)
- The "Tagline" and "Description" fields should be editable and the character length validation should work.
- Tapping on "Regenerate by AI" button should send the screen into loading mode.

## Screenshots

| Light | Dark |
|--------|--------|
| ![Simulator Screenshot - iPhone 14 Pro - 2024-01-03 at 17 39 44](https://github.com/woocommerce/woocommerce-ios/assets/524475/8b733c85-afb2-4d83-85b7-5b93704a86d2) | ![Simulator Screenshot - iPhone 14 Pro - 2024-01-03 at 17 39 52](https://github.com/woocommerce/woocommerce-ios/assets/524475/a6126a96-9950-435f-9e6a-44e995402617) | 

https://github.com/woocommerce/woocommerce-ios/assets/524475/20705b74-a3c7-4dcb-917a-d12e334dc369



---
